File: app/src/main/java/com/sbgapps/scoreit/app/ScoreItApp.java
package com.sbgapps.scoreit.app;

import android.app.Application;

/**
 * Created by sbaiget on 20/02/2017.
 */

public class ScoreItApp extends Application {

    private static GameManager sGameManager;

    public static GameManager getGameManager() {
        return sGameManager;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        init();
    }

    private void init() {
        if (null == sGameManager) sGameManager = new GameManager(getApplicationContext());
    }
}

File: app/src/main/kotlin/com/sbgapps/scoreit/app/ScoreItApp.kt
/*
 * Copyright 2019 Stéphane Baiget
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.sbgapps.scoreit.app

import android.app.Application
import com.jakewharton.threetenabp.AndroidThreeTen
import com.sbgapps.scoreit.app.di.uiModule
import com.sbgapps.scoreit.cache.di.cacheModule
import com.sbgapps.scoreit.data.di.dataModule
import org.koin.android.ext.koin.androidContext
import org.koin.core.context.startKoin
import timber.log.Timber

class ScoreItApp : Application() {

    override fun onCreate() {
        super.onCreate()

        AndroidThreeTen.init(this)
        if (BuildConfig.DEBUG) Timber.plant(Timber.DebugTree())

        startKoin {
            androidContext(this@ScoreItApp)
            modules(
                listOf(
                    uiModule,
                    dataModule,
                    cacheModule
                )
            )
        }
    }
}
